kubetest2/internal/deployers/eksapi/nodegroup.go

```diff
@@ -249,6 +249,23 @@ func (m *NodegroupManager) createUnmanagedNodegroupWithEFA(infra *Infrastructure
 	if err != nil {
 		return err
 	}
+	capacityReservations, err := m.clients.EC2().DescribeCapacityReservations(context.TODO(), &ec2.DescribeCapacityReservationsInput{})
+	if err != nil {
+		return fmt.Errorf("failed to describe capacity reservation")
+	}
+	var capacityReservationId string
+	if opts.CapacityReservation {
+		for _, cr := range capacityReservations.CapacityReservations {
+			if *cr.InstanceType == opts.InstanceTypes[0] && cr.State == "active" && *cr.AvailableInstanceCount >= int32(opts.Nodes) {
```

Comment:
I updated the PR to use a filter that retrieves capacity reservations, rather than using a for loop. I think it’s acceptable not to enforce that only a single instance type is passed; if multiple types are provided, we can select whichever is available.